db/migrate/20181113182615_add_vms_table.rb

```diff
@@ -0,0 +1,32 @@
+class AddVmsTable < ActiveRecord::Migration[5.1]
+  def change
+    create_table :vms do |t|
+      t.references :tenant, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+      t.references :source, :type => :bigint, :index => true, :null => false, :foreign_key => {:on_delete => :cascade}
+
+      t.string :source_ref
+      t.string :uid_ems
+      t.string :name
+      t.string :hostname
+      t.string :description
+      t.string :power_state
```

Comment:
Do we also need `raw_power_state`?

Reply:
Yeah I left that out because that sounds like something the parser should just set correctly, I don't know why we have both columns

Reply:
According to https://github.com/ManageIQ/manageiq/pull/481 it seems the action to take on the VM is determined by the power state from the provider.

Reply:
Looks like that is something that could use the `:extra` column on a per-provider basis e.g. openstack could store its raw_power_state there but we should never need to know that at the normalized level.